### PR TITLE
[DROOLS-6730] move drools-static from kogito to drools

### PR DESCRIPTION
**Pull request by mariofusco**
**JIRA**: https://issues.redhat.com/browse/DROOLS-6730

Related PRs:
https://github.com/kiegroup/drools/pull/4052
https://github.com/kiegroup/kogito-runtimes/pull/1763
https://github.com/kiegroup/optaplanner/pull/1700
https://github.com/kiegroup/kogito-apps/pull/1138

### CODE REVIEW

File: drools-core/pom.xml [hunk 1]
```diff
@@ -39,11 +39,11 @@
...
     <dependency>
       <groupId>org.drools</groupId>
-      <artifactId>drools-core-reflective</artifactId>
+      <artifactId>drools-wiring-api</artifactId>
     </dependency>
     <dependency>
       <groupId>org.drools</groupId>
-      <artifactId>drools-core-dynamic</artifactId>
+      <artifactId>drools-wiring-dynamic</artifactId>
       <scope>runtime</scope>
```
Comment (danielezonca) [inline drools-core/pom.xml:47]: Please open a ticket to refactor `drools` modules in kogito to use directly modules from here but to do that we need also to remove this runtime dependency: `drools-wiring-dynamic` must be added explicitly or using a aggregation module like `drools-engine`/`drools-engine-classic`

File: drools-wiring/drools-wiring-api/pom.xml [hunk 2]
```diff
@@ -4,19 +4,18 @@
...
         <version>8.15.0-SNAPSHOT</version>
-        <relativePath>../build-parent/pom.xml</relativePath>
     </parent>
 
     <groupId>org.drools</groupId>
-    <artifactId>drools-core-reflective</artifactId>
+    <artifactId>drools-wiring-api</artifactId>
 
-    <name>Drools :: Core :: Reflective</name>
+    <name>Drools :: Wiring :: API</name>
```
Comment (danielezonca) [inline drools-wiring/drools-wiring-api/pom.xml:15]: Please add a `description` to clarify the scope of this module

File: drools-wiring/drools-wiring-dynamic/pom.xml [hunk 3]
```diff
@@ -4,19 +4,18 @@
...
     <version>8.15.0-SNAPSHOT</version>
-    <relativePath>../build-parent/pom.xml</relativePath>
   </parent>
 
   <groupId>org.drools</groupId>
-  <artifactId>drools-core-dynamic</artifactId>
+  <artifactId>drools-wiring-dynamic</artifactId>
 
-  <name>Drools :: Core :: Dynamic</name>
+  <name>Drools :: Wiring :: Dynamic</name>
```
Comment (danielezonca) [inline drools-wiring/drools-wiring-dynamic/pom.xml:15]: Please add a `description` to clarify the scope of this module

File: drools-wiring/drools-wiring-static/pom.xml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  <parent>
+    <groupId>org.drools</groupId>
+    <artifactId>drools-wiring</artifactId>
+    <version>8.15.0-SNAPSHOT</version>
+  </parent>
+
+  <groupId>org.drools</groupId>
+  <artifactId>drools-wiring-static</artifactId>
+
+  <name>Drools :: Wiring :: Static</name>
```
Comment (danielezonca) [inline drools-wiring/drools-wiring-static/pom.xml:15]: Please add a `description` to clarify the scope of this module

File: drools-wiring/drools-wiring-static/src/main/java/org/drools/wiring/statics/StaticServiceRegistry.java [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+    private Map<Class<?>, Object> serviceMap = new HashMap<>();
+
+    private Map<String, Constructor<?>> constructorMap = new HashMap<>();
+
+    protected StaticServiceRegistry() {
+        wireServices();
+    }
+
+    protected void wireServices() {
```
Comment (danielezonca) [inline drools-wiring/drools-wiring-static/src/main/java/org/drools/wiring/statics/StaticServiceRegistry.java:48]: Is it possible to write a test that loads all `kie.conf` using dynamic and then verifies that the result map is the same of this static?

File: drools-wiring/pom.xml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.kie</groupId>
+    <artifactId>drools-build-parent</artifactId>
+    <version>8.15.0-SNAPSHOT</version>
+    <relativePath>../build-parent/pom.xml</relativePath>
+  </parent>
+
+  <name>Drools :: Wiring :: Parent</name>
```
Comment (danielezonca) [inline drools-wiring/pom.xml:13]: Can you add a `description`?

File: drools-wiring/drools-wiring-static/src/test/java/org/drools/wiring/statics/DummyByteArrayClassLoaderTest.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+public class DummyByteArrayClassLoaderTest {
+
+    @Test
+    public void defineClassShouldThrow() {
+        StaticComponentsSupplier.DummyByteArrayClassLoader cl =
+                new StaticComponentsSupplier.DummyByteArrayClassLoader();
+        assertThatThrownBy(() -> cl.defineClass("test", null, null))
+                .isInstanceOf(UnsupportedOperationException.class)
+                .hasMessageContaining("drools-core-static");
```
Comment (danielezonca) [inline drools-wiring/drools-wiring-static/src/test/java/org/drools/wiring/statics/DummyByteArrayClassLoaderTest.java]:
```suggestion
                .hasMessageContaining("drools-wiring-static");
```

File: drools-wiring/drools-wiring-static/src/main/java/org/drools/wiring/statics/StaticComponentsSupplier.java [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+        return new DummyByteArrayClassLoader();
+    }
+
+    public static class DummyByteArrayClassLoader implements ByteArrayClassLoader {
+        @Override
+        public Class<?> defineClass(final String name,
+                final byte[] bytes,
+                final ProtectionDomain domain) {
+            throw new UnsupportedOperationException(
+                    "The artifact drools-core-static does not support this operation, try using drools-core-dynamic instead.");
```
Comment (danielezonca) [inline drools-wiring/drools-wiring-static/src/main/java/org/drools/wiring/statics/StaticComponentsSupplier.java]:
```suggestion
                    "The artifact org.drools:drools-wiring-static does not support this operation, try using org.drools:drools-wiring-dynamic instead.");
```

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-drools/kie-pmml-models-drools-tree/kie-pmml-models-drools-tree-evaluator/src/test/java/org/kie/pmml/models/drools/tree/evaluator/implementations/HasKnowledgeBuilderMock.java [hunk 9]
```diff
@@ -18,13 +18,14 @@
...
-import org.drools.reflective.classloader.ProjectClassLoader;
+import org.drools.wiring.api.classloader.ProjectClassLoader;
 import org.kie.internal.builder.KnowledgeBuilder;
 import org.kie.memorycompiler.KieMemoryCompiler;
 import org.kie.pmml.api.exceptions.KiePMMLException;
 import org.kie.pmml.kie.dependencies.HasKnowledgeBuilder;
 
-public class HasKnowledgeBuilderMock implements HasKnowledgeBuilder {
+public class
+HasKnowledgeBuilderMock implements HasKnowledgeBuilder {
```
Comment (danielezonca) [inline kie-pmml-trusty/kie-pmml-models/kie-pmml-models-drools/kie-pmml-models-drools-tree/kie-pmml-models-drools-tree-evaluator/src/test/java/org/kie/pmml/models/drools/tree/evaluator/implementations/HasKnowledgeBuilderMock.java]: Revert

File: drools-core/pom.xml (hunk 1 above)
Comment (mariofusco) [reply drools-core/pom.xml:47]: https://issues.redhat.com/browse/KOGITO-6354

File: drools-wiring/drools-wiring-static/src/main/java/org/drools/wiring/statics/StaticServiceRegistry.java (hunk 5 above)
Comment (mariofusco) [reply drools-wiring/drools-wiring-static/src/main/java/org/drools/wiring/statics/StaticServiceRegistry.java:48]: https://issues.redhat.com/browse/DROOLS-6735

**Pull request merged**